plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/ComparePresenter.java

```diff
@@ -181,30 +248,57 @@ public void showCompareBetweenRevisions(final Path file,
 
     @Override
     public void onClose(final String newContent) {
-        if (!compareWithLatest || this.localContent == null || newContent.equals(localContent)) {
+        if (!isEdited(newContent)) {
             view.hide();
             return;
         }
 
-        ConfirmCallback confirmCallback = () -> comparedFile.updateContent(newContent)
-                                                            .then(ignored -> {
-                                                                final Container parent = comparedFile.getParent();
-
-                                                                if (parent != null) {
-                                                                    parent.synchronize();
-                                                                }
-
-                                                                eventBus.fireEvent(new FileContentUpdateEvent(comparedFile.getLocation()
-                                                                                                                          .toString()));
-                                                                view.hide();
-                                                            });
+        ConfirmCallback confirmCallback = () -> {
+            saveContent(newContent);
+            view.hide();
+        };
 
         CancelCallback cancelCallback = view::hide;
 
         dialogFactory.createConfirmDialog(locale.compareSaveTitle(), locale.compareSaveQuestion(), locale.buttonYes(), locale.buttonNo(),
                                           confirmCallback, cancelCallback).show();
     }
 
+    @Override
+    public void onSaveChangesClicked() {
+        if (compareWithLatest) {
+            view.getEditableContent(content -> {
+                if (isEdited(content)) {
+                    saveContent(content);
+                }
+            });
+        }
+    }
+
+    @Override
+    public void onNextDiffClicked() {
+        view.getEditableContent(content -> {
+            if (isEdited(content)) {
```

Comment:
Yes this is required. In case when the widget shows diff between two revisions a user cannot edit any content, so in case of comparison with not latest code no changes is supposed.